apps/src/music/player/MusicLibrary.ts

```diff
@@ -30,6 +33,36 @@ export default class MusicLibrary {
   getFolderForPath(path: string): SoundFolder | null {
     return this.groups[0].folders.find(folder => folder.path === path) || null;
   }
+
+  // A progression step might specify a smaller set of allowed sounds.
+  setAllowedSounds(allowedSounds : Sounds):void {
+    this.allowedSounds = allowedSounds;
+  }
+
+  // A sound picker might want to show the subset of sounds permitted by the
+  // progression's currently allowed sounds.
+  getAllowedSounds(folderType: string) : Sounds {
```

Comment:
Should our components be allowed to decide whether to use allowed sounds or not? I wonder if maybe we should just modify the model in place and mark certain sounds/folders `allowed: true/false`, and use those flags in the other functions here...not sure.

Also should we be calling this in `PatternPanel` and `ChordPanel` too?

Reply:
I really like the idea of having an immutable library that we initially load, and then being able to dynamically set the allowed sound list as we transition levels.  As part of that "functional" philosophy, I like having this transformation function which leaves the library unchanged but still retrieves the filtered set.

I suspect we might end up with places in our code that want access to the unfiltered library, and I think we trust our components to use the allowed set via this function when they should.

It seems most likely we'll want to expand this to patterns and chords too, but I figured we could add them in a follow-up if/when we get the ask. :)